source/fab/language/fortran.py

```diff
@@ -49,156 +99,165 @@ def __init__(self, database: StateDatabase):
             f'''create table if not exists fortran_unit (
                    id integer primary key,
                    unit character({self._FORTRAN_LABEL_LENGTH}) not null,
-                   filename character({FileInfoDatabase.PATH_LENGTH}) not null
+                   found_in character({FileInfoDatabase.PATH_LENGTH})
+                       references file_info (filename)
                    )''',
             '''create index if not exists idx_fortran_program_unit
-                   on fortran_unit(unit)''',
-            '''create index if not exists idx_fortran_filename
-                   on fortran_unit(filename)'''
+                   on fortran_unit (unit, found_in)'''
         ]
         self.execute(create_unit_table, {})
 
         # Although the current unit will already have been entered into the
         # database it is not necessarily unique. We may have multiple source
-        # files which define the same unit. Thus it can not be used as a
-        # foreign key.
+        # files which define similarly named units. Thus it can not be used as
```

Comment:
Is it _similarly_ or is it the same?  I know this is just a comment but I'm not sure we want the ambiguity (or maybe I'm wrong and "similar" is right - my impression was that we could have a `something_mod` in 2 or more different source files - is it subtly different to that?)

Reply:
I was using "similar" in its mathematic sense. i.e. identical. However your are right to point out that is not common usage. I will change it to something less ambiguous.